api/buckets/service.go

```diff
@@ -2003,6 +2025,47 @@ func (s *Service) PullPathAccessRoles(ctx context.Context, req *pb.PullPathAcces
 	}, nil
 }
 
+func (s *Service) DefaultArchiveConfig(ctx context.Context, req *pb.DefaultArchiveConfigRequest) (*pb.DefaultArchiveConfigResponse, error) {
+	log.Debug("received set default archive config request")
+
+	if !s.Buckets.IsArchivingEnabled() {
+		return nil, ErrArchivingFeatureDisabled
+	}
+
+	ba, err := s.Collections.BucketArchives.GetOrCreate(ctx, req.Key)
+	if err != nil {
+		return nil, fmt.Errorf("getting bucket archive data: %s", err)
+	}
+	archiveConfig := ba.DefaultArchiveConfig
+	if archiveConfig == nil {
+		// The queried BucketArchive was created before we added .DefaultArchiveConfig, so just return the default
+		archiveConfig = &defaultDefaultArchiveConfig
```

Comment:
Just use the default specified at the top of this file. Logic of not persisting a default on the `BucketArchive` struct is that maybe we want to change the default with time. Don't want to be stuck with an old default that was persisted to mongo. We should take this approach in Powergate @jsign .

Reply:
No very strong option here either. It made sense in my mind for `ArchiveConfig` just because it is a new concept and it feels good to not commit too much to some default that is persisted and force us into having to do a data migration soon when things change. We can discuss further for Powergate if it ever seems important.